src/web_interface/file_tree/file_tree.py

```diff
@@ -71,23 +108,42 @@ class FileTreeData(NamedTuple):
     included_files: Set[str]
 
 
-def get_correct_icon_for_mime(mime_type: Optional[str]) -> str:
+def get_icon_for_file(mime_type: Optional[str], file_name: str) -> str:
     '''
-    Retrieve the path to appropriate icon for a given mime type. The icons are located in the static folder of the
-    web interface and the paths therefore start with "/static". Archive types all receive the same icon.
+    Retrieve the path to the appropriate icon for a given mime type and file name. The icons are located in the static
+    folder of the web interface and the paths therefore start with "/static". Archive types all receive the same icon.
 
-    :param mime_type: The MIME type of a file (in the file tree).
+    :param mime_type: The MIME type of the file (in the file tree).
+    :param file_name: The file name.
```

Comment:
Like the comment above I'm confused why we need the filename.
From my understanding it should be like this:
If we don't know the mimetype figure it out by the filename.
Then use `get_icon(mimetype)` to get the icon?

Is there a reason we need the mimetype AND the filename?